gst/nnstreamer/nnstreamer_plugin_api_converter.h

```diff
@@ -0,0 +1,93 @@
+/**
+ * NNStreamer API for Tensor_Decoder Sub-Plugins
+ * Copyright (C) 2019 MyungJoo Ham <myungjoo.ham@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file  nnstreamer_plugin_api_converter.h
+ * @date  09 Dec 2019
+ * @brief Mandatory APIs for NNStreamer Converter sub-plugins (Need Gst Devel)
+ * @see https://github.com/nnsuite/nnstreamer
+ * @author  MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#ifndef __NNS_PLUGIN_API_CONVERTER_H__
+#define __NNS_PLUGIN_API_CONVERTER_H__
+
+#include "tensor_typedef.h"
+#include <gst/gst.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/***********************************************
+* External Converters                          *
+************************************************/
+
+/**
+ * @brief Converter's subplugin implementation.
+ */
+struct _NNStreamerExternalConverter {
+  const char *media_type_name;
+
+  /** 1. chain func, data handling. */
+  GstBuffer * (*convert) (GstTensorConverter * self,
```

Comment:
Question: is it necessary to pass the arg `GstTensorConverter * self` to external converter?

Reply:
Then, we may supply ```NNStreamerExternalConverter *this, GstTensorConfig *tensor_config /*(self->tensor_config)*/, some private-data``` or a struct containing these three instead.